### DISCUSSION TITLE
Python 3.9 support

### DISCUSSION THREAD

**Discussion by sergi0g** [mu-editor/mu#1698] [Q&A] [answered]
How can i update mu 1.0.3 python version to 3.9?

**Comment by carlosperate** [comment, marked as answer]
At the moment Mu is not compatible with Python 3.9 due to some dependencies having issues in some platforms.

You can run Mu from source code, and with a small code change, enable 3.9 as a compatible version. So at that point, whatever version of Python you use to `pip install -e ."[dev]"` will be the version of Python running Mu (in contrast, the installer package their own Python interpreter).

You can follow the dev instructions here to learn how to run Mu from source code: https://mu.readthedocs.io/en/latest/setup.html

And to enable Python 3.9 compatibility you can apply this patch: https://github.com/mu-editor/mu/pull/1272/commits/fa9fa19ec88a14e03660d58fbb415597ba0d98d2
